src/Chapter3.hs

```diff
@@ -971,6 +1086,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday
+   = Monday
+   | Tuesday
+   | Wednesday
+   | Thursday
+   | Friday
+   | Saturday
+   | Sunday
+   deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: Weekday -> Bool
+-- can this be written down more compact?
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
```

Comment:
You can write this function as:

```haskell
isWeekend day = day == Saturday || day == Sunday
```

or

```haskell
isWeekend day = day `elem` [Saturday, Sunday]
```

(but the last implementation will be slightly slower.